src/rvsdg/tests.rs

```diff
@@ -756,6 +756,183 @@ fn rvsdg_subst() {
     egraph.parse_and_run_program(EGGLOG_GAMMA_PROGRAM).unwrap();
 }
 
+#[test]
+fn rvsdg_subst_beneath() {
+    const EGGLOG_THETA_PROGRAM: &str = r#"
+    (let unsubsted
+        (Theta
+              (Node (PureOp (badd (BoolT) (Arg 1) (Arg 2))))
+              (VO (vec-of
+                (Node (PureOp (blt (BoolT) (Node (PureOp (badd (BoolT) (Arg 1) (Arg 2)))) (Arg 3))))
```

Comment:
I'm in favor of not let binding - since the program doesn't have a useful semantics and, and the functionality is pretty low-level, I think it's easiest to read if it's as explicit as possible.

It's not too bad with a diff checker:
https://www.diffchecker.com/gCfrJJFf/

Reply:
Okay, that's fine then